modules/gdscript/gdscript_parser.cpp

```diff
@@ -4157,6 +4162,39 @@ bool GDScriptParser::onready_annotation(AnnotationNode *p_annotation, Node *p_ta
 	return true;
 }
 
+bool GDScriptParser::tool_button_annotation(AnnotationNode *p_annotation, Node *p_node, ClassNode *p_class) {
+#ifdef TOOLS_ENABLED
+	AnnotationNode *annotation = const_cast<AnnotationNode *>(p_annotation);
+	FunctionNode *func = static_cast<FunctionNode *>(p_node);
+
+	if (!is_tool()) {
+		push_error("Tool buttons can only be used in tool scripts (add `@tool` to the top of the script).", p_annotation);
+		return false;
+	}
+
+	annotation->export_info.name = "__tool_button_" + func->identifier->name;
```

Comment:
```suggestion
	// Avoid name conflict.
	annotation->export_info.name = "@tool_button_" + func->identifier->name;
```

For consistency with groups. `@` makes the name to be an invalid identifier. This is not as important as the name in the `member_indices` hashmap (see the review comment to `gdscript_compiler.cpp`), but it still makes sense to have a unique name in the property list.